test/functorch/test_eager_transforms.py

```diff
@@ -1617,6 +1617,27 @@ def vmap(info, in_dims, input):
         with self.assertRaisesRegex(RuntimeError, "returned an incompatible"):
             result = vmap(Zeros.apply)(x)
 
+    def test_kwarg_only_tensors(self, device):
+        class MyClass(torch.autograd.Function):
```

Comment:
```
import torch

class MyClass(torch.autograd.Function):
    @staticmethod
    def forward(x, *, y):
        return x + y

    @staticmethod
    def setup_context(ctx, inputs, output):
        pass

    @staticmethod
    def vmap(info, in_dims, x, *, y):
        print(x, y)
        return x + y, 0

x = torch.randn(2, 3)
y = torch.randn(2, 3)

def f(x, y):
    return MyClass.apply(x, y=y)

result = torch.vmap(f)(x, y)
print(result)
```
What happens in the above? The problem with kwarg-only Tensors is that we don't unwrap the BatchedTensor that is y. So I think we should throw an error if we have any kwarg-only tensors. Thoughts?

Reply:
I see, that makes sense, thanks for the explanation! I've added a check for `torch.autograd.Function`. We'll get the error when running  the line `torch.vmap(f)(x, y)`.